 src/server/game/Handlers/CharacterHandler.cpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/server/game/Handlers/CharacterHandler.cpp b/src/server/game/Handlers/CharacterHandler.cpp
index 284a8eb..20c363d 100644
--- a/src/server/game/Handlers/CharacterHandler.cpp
+++ b/src/server/game/Handlers/CharacterHandler.cpp
@@ -1090,6 +1090,20 @@ void WorldSession::HandlePlayerLoginFromDB(LoginQueryHolder* holder)
     if (pCurrChar->HasAtLoginFlag(AT_LOGIN_FIRST)) {
         pCurrChar->RemoveAtLoginFlag(AT_LOGIN_FIRST);
 
+    if(pCurrChar->HasAtLoginFlag(AT_LOGIN_RENAME))
+    {
+        std::string nameUpdate_AT_LOGIN_UPDATE_NAME;
+        uint32 accID_Update_AT_LOGIN_UPDATE_NAME;
+        QueryResult result = CharacterDatabase.PQuery("SELECT name,account FROM characters WHERE `guid` = '%u'", pCurrChar->GetGUIDLow());
+        if(result)
+        {
+            nameUpdate_AT_LOGIN_UPDATE_NAME = (*result)[0].GetString();
+            accID_Update_AT_LOGIN_UPDATE_NAME = (*result)[1].GetUInt32();
+			sWorld->UpdateGlobalPlayerData(pCurrChar->GetGUIDLow(), PLAYER_UPDATE_DATA_NAME, nameUpdate_AT_LOGIN_UPDATE_NAME, pCurrChar->getLevel(), pCurrChar->getGender(),pCurrChar->getRace(), pCurrChar->getClass());
+			sWorld->AddGlobalPlayerData(pCurrChar->GetGUIDLow(), accID_Update_AT_LOGIN_UPDATE_NAME, nameUpdate_AT_LOGIN_UPDATE_NAME, pCurrChar->getGender(), pCurrChar->getRace(), pCurrChar->getClass(), pCurrChar->getLevel(), 0, 0);
+        }
+        pCurrChar->RemoveAtLoginFlag(AT_LOGIN_RENAME);
+    }
         sScriptMgr->OnFirstLogin(pCurrChar);
     }
 
